django_nameko/rpc.py

```diff
@@ -218,43 +225,41 @@ def get_pool(pool_name=None):
             raise ImproperlyConfigured('NAMEKO_CONFIG must be specified')
         NAMEKO_MULTI_POOL = [name for name in NAMEKO_CONFIG.keys() if name.islower()]
         # Lazy instantiation, acquire lock first to prevent dupication init
-        create_pool_lock.acquire()
-        if not nameko_global_pools:  # double check inside lock is importance
-            if NAMEKO_MULTI_POOL:
-                nameko_global_pools = dict()
-                if 'default' not in NAMEKO_CONFIG and 'AMQP_URL' not in NAMEKO_CONFIG['default']:
-                    raise ImproperlyConfigured(
-                        'NAMEKO_CONFIG must be specified and should include at least "default" config with "AMQP_URL"')
-                default_config = NAMEKO_CONFIG['default']
-                # default_context_data = NAMEKO_CONFIG['default']['POOL'].get('CONTEXT_DATA', dict())
-                # multi_context_data = getattr(settings, 'NAMEKO_MULTI_CONTEXT_DATA', dict())
-                for name, _config in NAMEKO_CONFIG.items():
-                    # each nameko_global_pools will have different config with default config as default
-                    if name != 'default':
-                        # overide default config with nameko_global_pools config by merging 2 dict
-                        pool_config = dict(mergedicts(default_config.copy(), _config))
-                    else:
-                        # default nameko_global_pools
-                        pool_config = default_config.copy()
-                    # extract nameko_global_pools config from RpcCluster config
-                    pool_size = pool_config.pop('POOL_SIZE', None)
-                    pool_context_data = pool_config.pop('POOL_CONTEXT_DATA', None)
-                    pool_timeout = pool_config.pop('POOL_TIMEOUT', 0)
-                    # init nameko_global_pools
-                    _pool = ClusterRpcProxyPool(pool_config, pool_size=pool_size, context_data=pool_context_data,
-                                                timeout=pool_timeout)
-                    _pool.start()
-                    # assign nameko_global_pools to corresponding name
-                    nameko_global_pools[name] = _pool
-            else:
-                # single nameko_global_pools with old style configuration
-                if not hasattr(settings, 'NAMEKO_CONFIG') or not settings.NAMEKO_CONFIG:
-                    raise ImproperlyConfigured(
-                        'NAMEKO_CONFIG must be specified and should include at least "AMQP_URL" key.')
-                nameko_global_pools = ClusterRpcProxyPool(settings.NAMEKO_CONFIG)
-                nameko_global_pools.start()  # start immediately
+        with create_pool_lock:
```

Comment:
suround with the contextmanager use of the lock. it prevent a bug: when the creation of the cluster raise a Exception, the lock.release was never called.